Add Enable-AdlStoreKeyVault cmdlet and help

begoldsm opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
There was a missing cmdlet to enable the keyvault for an ADLS account.
This cmdlet has been added, as well as a simple test for it and an
update to our help.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

begoldsm commented:
@markcowl any concerns with this PR? I want to make sure it makes it in for the next release.

markcowl commented on src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/EnableAzureRmDataLakeStoreKeyVault.cs:
```diff
@@ -0,0 +1,40 @@
+    {
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, Mandatory = true,
+            HelpMessage = "The Data Lake Store account to enable the Key Vault for")]
+        [ValidateNotNullOrEmpty]
```
Alias("Name")]

begoldsm replied:
This is a break from the pattern that we have been using for other ADLA and ADLS commands. Is it expected that we should include a "Name" alias as well now? I am ok with adding it here right now, just want to make sure that I know it should be added to the other cmdlets in the future as well.

begoldsm replied:
@markcowl just spoke with PM. We would rather hold off on this to make sure it makes sense for all of our cmdlets that have an Account/AccountName parameter, since we want to ensure a consistent parameter naming across all the ADL cmdlets for users. Given this, are you ok with accepting this PR without the Alias change?

markcowl replied:
@begoldsm This is just about piping scenarios.  If the intention is to allow piping from generic cmdlets, then this needs to be 'Name' to allow piping by PropertyName.  Otherwise, not sure why you would have piping ByPropertyName for the 'resourcegroupname' parameter

markcowl commented on src/ResourceManager/DataLakeStore/AzureRM.DataLakeStore.psd1:
```diff
@@ -82,6 +82,7 @@ CmdletsToExport = 'Get-AzureRmDataLakeStoreTrustedIdProvider',
                'Set-AzureRmDataLakeStoreFirewallRule', 
                'Add-AzureRmDataLakeStoreFirewallRule', 
                'Add-AzureRmDataLakeStoreItemContent', 
+			   'Enable-AzureRmDataLakeStoreKeyVault', 
```
indentation

markcowl commented:
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1613/

markcowl commented:
new on demand run: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1622/

begoldsm commented:
@markcowl I addressed your feedback and added the "name" alias. Now that we have an issue opened against this can we get this merged to ensure that it makes it in for the next release?

markcowl commented:
@begoldsm  You will need to fix the following static analysis issue from the on-demand run:

```"Enable-AzureRmDataLakeStoreKeyVault Does not support ShouldProcess but the cmdlet verb Enable indicates that it should.```

So you will either need to support ```ShouldProcess``` or add an exception if you are sure this cmdlet makes no permanent changes.

begoldsm commented:
@markcowl new on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1628/

begoldsm commented:
@markcowl  new NEW on demand run here (and it passed!): http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1631/
